lib/services/access_request.go

```diff
@@ -269,13 +262,31 @@ func CalculateAccessCapabilities(ctx context.Context, clock clockwork.Clock, clt
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
+	} else {
+		caps.ApplicableRolesForResources, err = v.applicableSearchAsRoles(ctx, req.ResourceIDs, req.Login)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
 	}
```

Comment:
Correct me if I'm wrong, but it's either `RequestableRoles` if the flag is set or `ApplicableRolesForResources` if no flag is set. The original condition was confusing.

Reply:
I was wrong! Had to revert that. Luckily it's covered by integration tests.